pilot/pkg/autoregistration/controller.go

```diff
@@ -206,43 +207,93 @@ func setConnectMeta(c *config.Config, controller string, conTime time.Time) {
 	delete(c.Annotations, DisconnectedAtAnnotation)
 }
 
+// RegisterWorkload determines whether a connecting proxy represents a non-Kubernetes
+// workload and, if that's the case, initiates special processing required for that type
+// of workloads, such as auto-registration, health status updates, etc.
+//
+// If connecting proxy represents a workload that is using auto-registration, it will
+// create a WorkloadEntry resource automatically and be ready to receive health status
+// updates.
+//
+// If connecting proxy represents a workload that is not using auto-registration,
+// the WorkloadEntry resource is expected to exist beforehand. Otherwise, no special
+// processing will be initiated, e.g. health status updates will be ignored.
 func (c *Controller) RegisterWorkload(proxy *model.Proxy, conTime time.Time) error {
-	if !features.WorkloadEntryAutoRegistration || c == nil {
+	if c == nil {
 		return nil
 	}
-	// check if the WE already exists, update the status
-	entryName := autoregisteredWorkloadEntryName(proxy)
+	var entryName string
+	var autoCreate bool
+	if features.WorkloadEntryAutoRegistration && proxy.Metadata.AutoRegisterGroup != "" {
+		entryName = autoregisteredWorkloadEntryName(proxy)
+		autoCreate = true
+	} else if features.WorkloadEntryHealthChecks && proxy.Metadata.WorkloadEntry != "" {
```

Comment:
where Metadata.WorkloadEntry is first set

Reply:
>The value of proxy.Metadata.WorkloadEntry comes from the ISTIO_META_WORKLOAD_ENTRY environment variable that a user can set for an Istio Proxy (SystemD Unit) on a VM.

Is it a must for healthcheck, right

I can understand the need to do healthcheck for WLE, but could not for setting the above Spec for users.

Reply:
Yes, `ISTIO_META_WORKLOAD_ENTRY` is used for the purposes of health checks.

When Istio Proxy (running on a VM) connects to `istiod` and starts sending health status updates, we need to infer a `WorkloadEntry` to save the health updates to.

In the case of auto-registration, `WorkloadEntry` name is generated based on metadata of the Istio Proxy.

In the case of externally created `WorkloadEntry`, we need to infer which `WorkloadEntry` corresponds to the connected Istio Proxy.

One solution would be to find `WorkloadEntry` by IP address. However, it wouldn't cover all the use cases:
* there might be multiple `WorkloadEntry`(s) per IP address
* `WorkloadEntry` might be defined for a different IP address than the one sent in the metadata of the Istio Proxy

That's why this PR proposes a solution where Istio Proxy must explicitly name the `WorkloadEntry` it corresponds to.